components/brave_today/browser/publishers_controller.h

```diff
@@ -36,6 +37,13 @@ class PublishersController {
     virtual void OnPublishersUpdated(PublishersController* controller) = 0;
   };
 
+  // The following methods return a pointer to a |Publisher|. This pointer is
+  // not safe to hold onto, as the object it points to will be destroyed when
+  // the publishers are updated (which happens regularly). If you need it to
+  // live longer, take a clone.
+  mojom::Publisher* GetPublisherForSite(const GURL& site_url);
+  mojom::Publisher* GetPublisherForFeed(const GURL& feed_url);
```

Comment:
why non-const?

Reply:
Had a read through of https://chromium.googlesource.com/chromium/src/+/refs/heads/main/styleguide/c++/const.md. Now both methods are `const` and return a const pointer. Cheers for the tip!